DateCase mistakenly matches uppercase AM/PM

Timestamps with uppercase AM/PM are matching this rule (which seems intended for lowercase letters).

Example:
```sh
$ vale <(echo '2022-10-27 05:34:32 PM')

 /dev/fd/14
 1:18  error  With lowercase letters, the     proselint.DateCase
              periods are standard.

✖ 1 error, 0 warnings and 0 suggestions in 1 file.
```

--- a/proselint/DateCase.yml
+++ b/proselint/DateCase.yml
@@ -1,6 +1,6 @@
 extends: existence
 message: With lowercase letters, the periods are standard.
-ignorecase: true
+ignorecase: false
 level: error
 nonword: true
 tokens:

